dtfs-central-api/src/v1/controllers/utilisation-report-service/helpers/keying-sheet-mark-as-helpers.ts

```diff
@@ -0,0 +1,22 @@
+import { FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { NotFoundError } from '../../../../errors';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+
+export const getSelectedFeeRecordsAndUtilisationReportForKeyingDataMarkAs = async (
+  reportId: number,
+  selectedFeeRecordIds: number[],
+): Promise<{ selectedFeeRecords: FeeRecordEntity[]; utilisationReport: UtilisationReportEntity }> => {
+  const selectedFeeRecords = await FeeRecordRepo.findByIdAndReportIdWithReport(selectedFeeRecordIds, reportId);
+
+  if (selectedFeeRecords.length !== selectedFeeRecordIds.length) {
+    throw new NotFoundError(`Could not find report with id ${reportId} and with fee records with ids ${selectedFeeRecordIds.join(', ')}`);
+  }
+
+  const utilisationReport = selectedFeeRecords[0]?.report;
+
+  if (!utilisationReport) {
+    throw new NotFoundError(`Could not find report with id ${reportId} and with fee records with ids ${selectedFeeRecordIds.join(', ')}`);
```

Comment:
This is more of a semantic thing, but I technically think this is a server error rather than a not found error - if we have found fee records, the report should exist on it and, if it doesn't, that means something unexpected happened. Could we possibly just throw a normal `Error` so that the logs show a `500` and have a different error message?